File: workflow/plugins/python/logic/logic_gt/logic_gt.py
"""Workflow plugin: greater than comparison."""

from ...base import NodeExecutor


class LogicGt(NodeExecutor):
    """Check if a > b."""

    node_type = "logic.gt"
    category = "logic"
    description = "Check if a > b"

    def execute(self, inputs, runtime=None):
        a = inputs.get("a")
        b = inputs.get("b")
        return {"result": a > b}

File: workflow/plugins/go/logic/logic_gt/logic_gt.go
// Package logic_gt provides a workflow plugin for greater than comparisons.
package logic_gt

// LogicGt implements the NodeExecutor interface for greater than comparisons.
type LogicGt struct {
	NodeType    string
	Category    string
	Description string
}

// NewLogicGt creates a new LogicGt instance.
func NewLogicGt() *LogicGt {
	return &LogicGt{
		NodeType:    "logic.gt",
		Category:    "logic",
		Description: "Check if a is greater than b",
	}
}

// Execute runs the plugin logic.
func (p *LogicGt) Execute(inputs map[string]interface{}, runtime interface{}) map[string]interface{} {
	a := toFloat(inputs["a"])
	b := toFloat(inputs["b"])

	return map[string]interface{}{"result": a > b}
}

func toFloat(v interface{}) float64 {
	switch n := v.(type) {
	case float64:
		return n
	case int:
		return float64(n)
	case int64:
		return float64(n)
	default:
		return 0
	}
}
